lib/src/main/java/com/scalar/admin/kubernetes/Pauser.java

```diff
@@ -123,17 +149,33 @@ private void unpauseWithRetry(
         return;
       } catch (Exception e) {
         if (++retryCounter >= maxRetryCount) {
-          logger.warn(
+          // If someone uses this library directly instead of using our CLI, users should handle the
+          // exception properly. However, this case is a very critical issue. Therefore, we output
+          // the error message here despite whether the exception is handled or not on the caller
+          // side.
+          logger.error(
               "Failed to unpause Scalar product. They are still in paused. You must restart related"
                   + " pods by using the `kubectl rollout restart deployment {}`"
-                  + " command to unpase all pods.",
+                  + " command to unpause all pods.",
               target.getDeployment().getMetadata().getName());
-          return;
+          // In our CLI, we catch this exception and output the message as an error on the CLI side.
+          throw new PauserException(
+              String.format(
+                  "Failed to unpause Scalar product. They are still in paused. You must restart"
+                      + " related pods by using the `kubectl rollout restart deployment %s` command"
+                      + " to unpause all pods.",
+                  target.getDeployment().getMetadata().getName()),
+              e);
         }
       }
     }
   }
 
+  @VisibleForTesting
+  TargetSnapshot getTarget() throws PauserException {
+    return targetSelector.select();
+  }
```

Comment:
As I mentioned in another comment, I separated this `targetSelector.select()` as one method `getTarget()` for testing.